Add `--schema` to the neo4j-admin import tool (#1822)
neo4j/docs-operations: modules/ROOT/pages/tools/neo4j-admin/neo4j-admin-import.adoc

====
Indexes and constraints are not created during the import.
Instead, you have to add these afterward (see link:{neo4j-docs-base-uri}/cypher-manual/{page-version}/indexes-for-full-text-search[Cypher Manual -> Indexes]).

> Starting from Neo4j 5.24, you can use the `--schema` option to create indexes and contraints during the import process.

Author: @docrozza, is this option for the community edition?

Reviewer: It only works for block format - so kinda enterprise by default
Also, only FULL at the moment but it will be available for incremental at some point (still enterprise though)

> Starting from Neo4j 5.24, you can use the `--schema` option to create indexes and contraints during the import process.

Author: ```suggestion
Starting from Neo4j 5.24, you can use the `--schema` option to create indexes and contraints during the import process.
The option is available in the Enterprise Edition and works only for the block format.
```

                                 [--bad-tolerance=<num>] [--delimiter=<char>] [--format=<format>]
                                 [--high-parallel-io=on|off|auto] [--id-type=string|integer|actual]
                                 [--input-encoding=<character-set>] [--max-off-heap-memory=<size>] [--quote=<char>]
                                 [--schema=<path>]
                                 [--read-buffer-size=<size>] [--report-file=<path>] [--threads=<num>] --nodes=[<label>[:
                                 <label>]...=]<files>... [--nodes=[<label>[:<label>]...=]<files>...]...

>                                  [--relationships=[<type>=]<files>...]... <database>

Reviewer: from the build failure does `--schema` need to be included somewhere here too?

Author: yes, `--schema` has to be included. But I don't understand where to put it to make our test happy.
Also, if I read the test's results correctly,  the `--schema` option must be in the incremental part too.

Reviewer: It's a weird one - it needs to be in the basic command so it to can be picked up by all 3 import types (full, incremental and SPDY) - even in community runs
Order wise, I think they're alphabetical so after --relationships?

This is done by using the `--verbose` option.
|import.report

> |--schema=<path> label:new[Introduced in 5.24] label

Author: ```suggestion
|--schema=<path> label:new[Introduced in 5.24] label:enterprise[Enterprise edition]
```

> |File in which to store the Cypher schema commands to run as part of the data import.

Author: ```suggestion
|Path to the file containing the Cypher schema commands for creating indexes and constraints as part of the data import.
```

--relationships import/roles_header.csv,import/roles.csv
----

[[indexes-constraints-import]]
==== Provide indexes and constraints during import

Starting with Neo4j 5.24, you can use the `--schema` option that allows Cypher commands to be provided to create indexes/constraints during the initial import process.
It currently only works for full import.

You should have a Cypher script containing only `CREATE INDEX|CONSTRAINT` commands to be parsed and executed.
This file uses ';' as the separator.

List of supported indexes and constraints that can be created by the import tool:

* RANGE

> * LOOKUP

Author: ```suggestion
* LOOKUP
* POINT
* TEXT
```

* FULL-TEXT

> * VECTOR

Reviewer: Also POINT and TEXT

For example:

[source, shell, role=noplay]
----
bin/neo4j-admin database import full neo4j --nodes=import/movies.csv --nodes=import/actors.csv --relationships=import/roles.csv --schema=import/schema.cypher 

> ----

Reviewer: Suggestion: include an example for the `import/schema.cypher` like
```
CREATE INDEX PersonNameIndex FOR (i:Person) ON (i.name);
CREATE CONSTRAINT PersonAgeConstraint FOR (c:Person) REQUIRE c.age IS :: INTEGER
```

--relationships import/roles_header.csv,import/roles.csv
----

> [[indexes-constraints-import]]

Author: ```suggestion
[role=label--enterprise-edition label--new-5.24]
[[indexes-constraints-import]]
```

==== Provide indexes and constraints during import

Starting with Neo4j 5.24, you can use the `--schema` option that allows Cypher commands to be provided to create indexes/constraints during the initial import process.

> It currently only works for full import.

Author: ```suggestion
It currently only works for the block format and full import.
```
